onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2726,15 +2726,73 @@ def aten_index_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::index_put")
 def aten_index_put(
-    self: TensorType,
-    indices: Optional[Sequence[TensorType]],
-    values: TensorType,
+    self: TReal,
+    indices: Sequence[TReal],
+    values: TReal,
     accumulate: bool = False,
-) -> TensorType:
+) -> TReal:
     """index_put(Tensor self, Tensor?[] indices, Tensor values, bool accumulate=False) -> Tensor"""
 
-    raise NotImplementedError()
+    index = op.SequenceAt(indices, 0)  # assume indices only have 1 element
+    # change array([1,3]) to array([[1,1,1,1,1],[3,3,3,3,3]])
+    self_rank_1 = op.Gather(op.Shape(self), 1)
+    index_rank_0 = op.Gather(op.Shape(index), 0)
+    neg_1 = op.Constant(value_ints=[-1])
+    shape = op.Concat(op.Reshape(self_rank_1, neg_1), op.Reshape(index_rank_0, neg_1), axis=0)
+    new_ind = op.Expand(index, shape)
+    new_ind_t = op.Transpose(new_ind)
+
+    if accumulate:
```

Comment:
This is a little tricky: We need to cast accumulate to BOOL when the attribute is used as a condition in if statements. This is because the attribute is internally represented as int, and becomes INT64 when implicitly casted to Input when participating in the logic. We may be able to update onnx script to do this cast automatically? @gramalingam

Reply:
changed  to op.Cast(to=INT64.dtype)